src/components/PopupWithForm.js

```diff
@@ -23,11 +23,23 @@ class PopupWithForm extends Popup {
     super.setEventListeners();
     this._popupForm.addEventListener("submit", (evt) => {
       evt.preventDefault();
-      this._handleFormSubmit(this._getInputValues());
-      this._popupForm.reset();
-      this.close();
+      this._handleFormSubmit(this._getInputValues(), this._button);
     });
   }
+
+  close() {
+    super.close();
+    this._popupForm.reset(); // Reset the form fields when closing
+  }
+
```

Comment:
**COULD BE IMPROVED**

You can make a special method for getting a form element from `PopupWithForm` not to **search** it in `index.js`



```js

  getForm() {
    return this._popupForm
  }


```